xwakes/wit/component.py

```diff
@@ -385,3 +511,944 @@ def get_shorthand_type(self) -> str:
+                (plane == 'y' and exponents == (0, 1, 0, 0))):
+            out = factor * length * ((1 + np.sign(f)*1j) * material_resistivity
+                            / (np.pi * radius**3 * (2 * np.pi * f * np.sqrt(eps0 * mu0)))
+                            / self.delta_skin(f, material_resistivity,
+                                              material_magnetic_permeability))
+        else:
+            raise ValueError("Resistive wall wake not implemented for "
+                  "component {}{}. Set to zero".format(plane, exponents))
+
+        return out
+
+    def wake(self, t, beta0=None):
+        layer = self.layer
+        if layer is not None:
+            self._check_layer(layer) # Checks that there are no unsupported layer properties
+                                # (e.g. permeability different from vacuum)
+            material_resistivity = layer.dc_resistivity
+        else:
+            material_resistivity = self.resistivity
+        radius = self.radius
+        plane = self.plane
+        factor = self.factor
+        length = self.length
+        exponents = tuple(self.source_exponents + self.test_exponents)
+
+        beta0 = beta0 if beta0 is not None else 1
+
+        isscalar = np.isscalar(t)
+        t = np.atleast_1d(t)
+        mask_positive = t > 1e-20
+        out = np.zeros_like(t)
+
+        # Longitudinal
+        if plane == 'z' and exponents == (0, 0, 0, 0):
+            out[mask_positive] = factor * length * (-1. / (4 * np.pi * radius)
+                            * np.sqrt(Z0 * material_resistivity / np.pi / c_light)
+                            * 1/(t[mask_positive]**(3/2)))
+        # Transverse dipolar
+        elif ((plane == 'x' and exponents == (1, 0, 0, 0)) or
+                (plane == 'y' and exponents == (0, 1, 0, 0))):
+            out[mask_positive] = factor * length * (
+                        1. / (np.pi * radius**3)
+                        * np.sqrt(c_light * Z0 * material_resistivity / np.pi)
+                        * 1/(t[mask_positive]**(1/2)))
+        else:
+            raise ValueError("Resistive wall wake not implemented for "
+                  "component {}{}. Set to zero".format(plane, exponents))
+
+        if isscalar:
+            out = out[0]
+
+        return out
+
+    @staticmethod
+    def delta_skin(f, material_resistivity, material_magnetic_permeability):
+        return np.sqrt(2 * material_resistivity / (2*np.pi*abs(f) *
+                                        material_magnetic_permeability))
+
+    @property
+    def f_rois(self):
+        return []
+
+    @property
+    def t_rois(self):
+        return []
+
+    def function_vs_t(self, t, beta0, dt):
+
+        isscalar = np.isscalar(t)
+        t = np.atleast_1d(t)
+
+        assert dt > 0
+        mask_zero = np.abs(t) < dt * self.zero_rel_tol
+        out = np.zeros_like(t)
+        out[mask_zero] = self.wake(dt * self.zero_rel_tol, beta0=beta0)
```

Comment:
What we are doing here is a sort to bound the otherwise unbounded resistive wall wake. For any time smaller than `dt*self.zero_rel_tol` we impose that the wake is equal to the one in `dt*self.zero_rel_tol`. This is very similar to what was done in `PyHEADTAIL`. I think in principle one could set it to zero but we have to see how the impedance of such a wake would look compared to the analytic thick wall impedance.

Reply:
I agree - my bad, I got confused somehow.